FreeRTOS/Test/CMock/smp/multiple_priorities_preemption_disable/multiple_priorities_preemption_disable_utest.c

```diff
@@ -179,3 +179,978 @@ int suiteTearDown( int numFailures )
+     * task  did not go to sleep, technically nothing happens */
+}
+
+/**
+ * @brief This test ensures that when we call prvIdleTask and the ready tasks
+ *        lists contains 1 elemets and  the top ready priority is less or equal
+ *        to the idle priority, then we let the suggested time to sleep is
+ *        returned
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ * prvGetExpectedIdleTime();
+ *
+ * else if( listCURRENT_LIST_LENGTH( &( pxReadyTasksLists[ tskIDLE_PRIORITY ] ) ) > 1 )
+ *
+ * @endcode
+ *
+ * configNMBER_OF_CORES > 1
+ * configUSE_TICKLESS_IDLE != 0
+ * configUSE_PORT_OPTIMISED_TASK_SELECTION = 0
+ */
+void test_prvGetExpectedIdleTime_ready_list_eq_1( void )
+{
+    CEXCEPTION_T e = CEXCEPTION_NONE;
+    TCB_t xTCB = { 0 };
+
+    xTickCount = 230;
+    xNextTaskUnblockTime = 240; /* expectedidletime = xNextTaskUnblockTime - xTickCount */
+    pxCurrentTCBs[ 0 ] = &xTCB;
+    xTCB.uxPriority = tskIDLE_PRIORITY;
+
+    /* Test Setup */
+    uxDeletedTasksWaitingCleanUp = 0;
+    uxTopReadyPriority = tskIDLE_PRIORITY;
+
+    /* Test Expectations */
+    vFakePortYield_Expect();
+    listCURRENT_LIST_LENGTH_ExpectAnyArgsAndReturn( 1 );
+
+    ulFakePortSetInterruptMask_ExpectAndReturn( 0 );
+    vFakePortGetCoreID_ExpectAndReturn( 0 );
+    vFakePortClearInterruptMask_Expect( 0 );
+
+    listCURRENT_LIST_LENGTH_ExpectAnyArgsAndReturn( 1 );
+
+    /* vTaskSuspendAll */
+    vFakePortAssertIfISR_Expect();
+    ulFakePortSetInterruptMask_ExpectAndReturn( 0 );
+    vFakePortGetTaskLock_Expect();
+    vFakePortGetISRLock_Expect();
+    vFakePortReleaseISRLock_Expect();
+    vFakePortClearInterruptMask_Expect( 0 );
+
+
+    ulFakePortSetInterruptMask_ExpectAndReturn( 0 );
+    vFakePortGetCoreID_ExpectAndReturn( 0 );
+    vFakePortClearInterruptMask_Expect( 0 );
+
+    listCURRENT_LIST_LENGTH_ExpectAnyArgsAndReturn( 1 );
+
+    /* Test Verifications */
+
+    /* this function being called is the aim of this test, it proves that the
+     * task went to sleep the specified amount of time. */
+    vPortSuppressTicksAndSleep_Expect( xNextTaskUnblockTime - xTickCount );
+
+    vFakePortEnterCriticalSection_Expect();
+    vFakePortGetCoreID_ExpectAndReturn( 0 );
+    vFakePortReleaseTaskLock_Expect();
+    vFakePortExitCriticalSection_Expect();
+
+    listCURRENT_LIST_LENGTH_ExpectAndThrow( &( pxReadyTasksLists[ tskIDLE_PRIORITY ] ),
+                                            EXIT_LOOP );
+
+
+    /* API Call */
+    portTASK_FUNCTION( prvIdleTask, args );
+
+    Try
+    {
+        prvIdleTask( NULL );
+    }
+    Catch( e )
+    {
+        if( e == EXIT_LOOP )
+        {
+            TEST_PASS();
+        }
+        else
+        {
+            TEST_FAIL();
+        }
+    }
+    /* Test Verifications */
+    /* the verification of the test is above in the expectations */
+}
+
+void port_assert_if_isr_cb( int num_callbacks )
```

Comment:
This function is used in the test case only. Suggest we use static function here and move it to helper function section.